Editor/Scripts/PackageUpdate/PackageUpdateHandler.cs

```diff
@@ -0,0 +1,58 @@
+using System;
+using System.Threading;
+using Unity.FilmInternalUtilities.Editor;
+using UnityEditor;
+using UnityEditor.PackageManager;
+using UnityEditor.SceneManagement;
+using UnityEngine;
+using Object = UnityEngine.Object;
+
+
+namespace Unity.MeshSync.Editor{
+
+[InitializeOnLoad]
+internal class PackageUpdateHandler {
+    static PackageUpdateHandler() {
+        Events.registeredPackages -= OnPackageRegistered;
+        Events.registeredPackages += OnPackageRegistered;
+
+        Events.registeringPackages -= OnPackageRegistering;
+        Events.registeringPackages += OnPackageRegistering;
+    }
+
+    private static void OnPackageRegistering(PackageRegistrationEventArgs obj) {
+        if (obj.changedFrom.FindPackage("com.unity.meshsync") == null)
+            return;
+
+        StopAllServers();
+    }
+
+    private static void OnPackageRegistered(PackageRegistrationEventArgs obj) {
+
+        if (obj.changedTo.FindPackage("com.unity.meshsync") == null)
+            return;
+           
+        // If the package was just updated, stop all servers and restart the editor
+        StopAllServers();        
+
+        // Save the current open scenes
+        EditorSceneManager.SaveOpenScenes();
```

Comment:
I am wondering if it's wise to forcefully save opened scenes without asking users.

In a related note, there is actually code to ask user to restart the project after updating MeshSync package because Unity can't unload previously loaded DLLs (previous MeshSync version).
This code is in PackageEventSubscriber class, and is not active at the moment due to wrong ifdef, but I'll activate it again.

So, what if we just simply stop editor servers and ask user to restart Unity after they have updated MeshSync package ?

Reply:
> I am wondering if it's wise to forcefully save opened scenes without asking users.

I have the same concern. The other option as you mentioned, is to ask users to restart, how do we deal with the user manually starting the server? we could show a pop-up when they click on any of the start buttons, notifying them that they cannot use the server and ask them to restart. What do you think?

Reply:
> In a related note, there is actually code to ask user to restart the project after updating MeshSync package because Unity can't unload previously loaded DLLs (previous MeshSync version).

So we should be restarting after update, in any case right? My fear was that this solution introduces friction for solving the hanging issue, but sounds like it's good practice?

Reply:
> So we should be restarting after update, in any case right?

Exactly.

> we could show a pop-up when they click on any of the start buttons, notifying them that they cannot use the server and ask them to restart. What do you think?

It's better, but not necessary since there should already be a dialog box that asks them to restart Unity after they upgraded MeshSync.